angular2/src/angular2TextMask.ts

```diff
@@ -33,6 +33,7 @@ export default class MaskedInputDirective {
   }
 
   writeValue(value: any) {
+    this.textMaskInputElement.update(value)
```

Comment:
Ya, it's not ideal. Not sure how to work around it. `writeValue` is getting called immediately before anything even happens so the textInput was getting a blank value loaded. I'm not sure why that's happening to begin with though, seems it should be getting the proper values from the models but there's obviously something about this `ControlValueAccessor` we're not understanding. I'll ponder it some more tomorrow.